pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -428,6 +438,25 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    logBrokerResponse(requestStatistics, requestId, query, compilationStartTimeNs, brokerRequest,
+        numUnavailableSegments, serverStats, brokerResponse, executionEndTimeNs);
+    return brokerResponse;
+  }
+
+  /**
+   * Given a {@link BrokerRequest}, this function will determine if we can return a response without server-side query
+   * evaluation. This happens when the optimizer determines that the entire WHERE clause evaluates to false.
+   */
+  private boolean isResponsePossible(BrokerRequest brokerRequest) {
```

Comment:
Suggest renaming to `isAlwaysFalse`

Reply:
Renamed to `isFilterAlwaysFalse` and also added 'isFilterAlwaysTrue' to take care of all cases including setting `offlineBrokerRequest` or `realtimeBrokerRequest` to null if needed and dropping filter if filter will always evaluate to TRUE.